### PR TITLE
Fix Image block is not displaying the image on Project Renderer

### PR DESCRIPTION
**Pull request by jcheringer**
## Summary

The purpose of this PR is to fix an issue regarding the images from Image Block not being rendered on Project Renderer view.

Ref: c/szZ0mf3V-tr

## Test Instructions
 
* Checkout this branch
* Run `yarn install` and start the dashboard using `yarn workspace @crowdsignal/dashboard start`
* Create or open a project, add an Image Block and set the image properly.
* Start the Project Renderer using `yarn workspace @crowdsignal/project-renderer start`
* Open the project page inside the Renderer (usually: https://crowdsignal.localhost:9001/[projectId])
* You should see the Image Block rendering the image correctly.

### CODE REVIEW

File: packages/blocks/src/render-blocks.js [hunk 1]
```diff
@@ -109,21 +109,26 @@ const appendInnerBlocks = ( blockName, elements, innerBlocks ) => {
...
+			...createElementsRecursive( element.children ),
+		];
+
 		return createElement(
 			element.name,
 			element.props,
 
 			// In practice, if an element is the host for inner blocks
 			// it's children should alwyas be empty here.
-			element.innerBlocks || createElementsRecursive( element.children )
```
Comment (jcheringer) [inline packages/blocks/src/render-blocks.js:126]: While debugging the code I found out that, for the block `core/cover` both `element.innerBlocks` and `element.children` are populated with some objects.

This is preventing the `` from being added to the elements list.

Changing this solved the issue for the image but, if the Image Block has some text over it, the created `<p>` element is being added to the elements list before the `` and, when rendered, the text gets hidden behind the image.

Took a look at WP.com behavior and there, the `core/cover` creates a wrapper with the class `wp-block-cover__inner-container` and adds the `<p>` element inside the wrapper.

Still didn't figure out why this is not happening on our side.

**Review by ice9js — COMMENTED** [review]
So having looked at it.. I think I see what the problem might be and we actually have a solution ready but it hasn't been used yet.

The core problem is, for some blocks, `core/cover` being a good example, generates a lot of markup in `originalContent` which makes it very difficult to guess where any children/innerblocks should go compared to simple blocks which just have one node.

For example, this is how `originalContent` looks for one of my cover blocks:

```html
<div class=\"wp-block-cover is-light\"><span aria-hidden=\"true\" class=\"wp-block-cover__gradient-background has-background-dim\"></span><img class=\"wp-block-cover__image-background\" alt=\"\" src=\"https://oplatkinatort.pl/wp-content/uploads/2015/03/oplatek-oplatki-na-tort-pinkie-pie.jpg\" data-object-fit=\"cover\"/><div class=\"wp-block-cover__inner-container\"></div></div>
```

Compared to something like a paragraph block which has just: `<p></p>`.

This is a problem that needs to be solved for each block individually which is why we added [that switch statement in appendInnerBlocks](https://github.com/Automattic/crowdsignal-ui/blob/main/packages/blocks/src/render-blocks.js#L89).

Whenever we encounter something like this, that's the place where we should add rules for inserting `innerBlocks` in the correct node depending on the blocks.  

>Took a look at WP.com behavior and there, the core/cover creates a wrapper with the class wp-block-cover__inner-container and adds the <p> element inside the wrapper.
>
>Still didn't figure out why this is not happening on our side.

WordPress renders all its blocks on the backend for public views, and thus doesn't have such issues. In our case, we rely on `originalContent` to carry over the markup from the editor to our public views.

**Comment by jcheringer** [comment]
Hey @ice9js, thanks for the explanation!
I did the changes to handle the `core/cover` block inside the switch statement and it seems to work fine. 👏 
Could you take a look again?

File: packages/blocks/src/render-blocks.js [hunk 2]
```diff
@@ -95,7 +96,16 @@ const appendInnerBlocks = ( blockName, elements, innerBlocks ) => {
 
 		// 	container.innerBlocks = innerBlocks;
 		// 	return;
```
Comment (ice9js) [inline packages/blocks/src/render-blocks.js:98]: Nit: could we remove this commented code while we're at it? It was here only as a reference but now that we have a working example, I think we can just get rid of it.

**Review by ice9js — APPROVED** [review]
Seems to work for me, thanks! Let's maybe just removed the old commented-out example before merging.

File: packages/blocks/src/render-blocks.js (hunk 2 above)
Comment (jcheringer) [reply packages/blocks/src/render-blocks.js:98]: Sure!

**Pull request merged**